jni/src/main/java/com/horizen/merkletreenative/BigLazyMerkleTree.java

```diff
@@ -28,15 +30,15 @@ public static BigLazyMerkleTree init(int height, String statePath, String dbPath
     // Returns the position to which insert the leaf
     public long getPosition(FieldElement leaf) {
         if (lazyMerkleTreePointer == 0)
-            throw new IllegalArgumentException("lazyMerkleTreePointer must be not null.");
+            throw new IllegalArgumentException("LazyBigMerkleTree instance was freed.");
```

Comment:
I would suggest to change `IllegalArgumentException` to `IllegalStateException` type. Same bellow.